### PR TITLE
Lize fix tests

### PR DESCRIPTION
**Pull request by lvdheever**
@theunsa Please review carefully. I think there were some potential problems in the Container set_sensor_sampling and set_sensor_listener and I reworked both.

Also moved the two failing tests (test_set_sensor_sampling and test_set_sensor_listener) to the integrated container test class so that the test covers more and use less mocks.  Please also review the tests to see that it correctly test those two functions.

### CODE REVIEW

File: katcp/resource_client.py [hunk 1]
```diff
@@ -881,33 +881,37 @@ def list_sensors(self, filter="", strategy=False, status="",
...
+        # Otherwise, depend on self._add_sensors() to handle it when the sensor appears
 
     @steal_docstring_from(resource.KATCPResource.set_sensor_listener)
     def set_sensor_listener(self, sensor_name, listener):
         sensor_name = resource.escape_name(sensor_name)
         sensor_obj = getattr(self.sensor, sensor_name, None)
-        for child_name in dict.keys(self.children):
+        if sensor_obj:
+            # The sensor exists, so register the listener.
+            child_name = sensor_obj.parent_name
```
Comment (ludwigschwardt) [inline katcp/resource_client.py]: The parent is a child too? :-)

Comment (lvdheever) [inline katcp/resource_client.py]: Yes - the parent of sensor_obj is a child of the container.
Will see if there is some rework that can be done on the naming.

File: katcp/resource_client.py [hunk 2]
```diff
@@ -881,33 +881,37 @@ def list_sensors(self, filter="", strategy=False, status="",
     def set_sensor_strategy(self, sensor_name, strategy_and_parms):
         sensor_name = resource.escape_name(sensor_name)
         sensor_obj = getattr(self.sensor, sensor_name, None)
-        for child_name in dict.keys(self.children):
+        if sensor_obj:
+            # The sensor exists, so set the strategy.
+            child_name = sensor_obj.parent_name
```
Comment (theunsa) [inline katcp/resource_client.py]: I agree with @ludwigschwardt that we should distinguish who's-child are we manipulating here.

**Comment by theunsa** [comment]
Rest is fine.

**Comment by theunsa** [comment]
@lvdheever: Just a ping ... if we call that variable something more accurate I'll merge ... does not necessarily have to happen everywhere ... at least for the new code.

**Comment by lvdheever** [comment]
Renaming done.

**Pull request merged**